src/Engine/ProtoCore/Lang/BuiltInMethods.cs

```diff
@@ -807,7 +815,35 @@ public BuiltInMethods(Core core)
                     {
                         new KeyValuePair<string, Type>("list", TypeSystem.BuildPrimitiveTypeObject(PrimitiveType.Var)),
                     }.ToList(),
-                    ID = BuiltInMethods.MethodID.ToStringFromArray
+                    ID = BuiltInMethods.MethodID.ToStringFromArray,
+                    MethodAttributes = new MethodAttributes(true, false, @"This method is obsolete, please use 'ToStringFromArrayAndFormat' "),
+                },
+
+                new BuiltInMethod
+                {
+                    ReturnType = TypeSystem.BuildPrimitiveTypeObject(PrimitiveType.String, 0),
+                    Parameters = new List<KeyValuePair<string, ProtoCore.Type>>
+                    {
+                        new KeyValuePair<string, ProtoCore.Type>("object", TypeSystem.BuildPrimitiveTypeObject(PrimitiveType.Var, 0)),
+                        new KeyValuePair<string, ProtoCore.Type>("formatSpecifier", TypeSystem.BuildPrimitiveTypeObject(PrimitiveType.String, 0)),
+                        //TODO (MJK)
+                        //if we wanted to support default args for builtins we would need to support these names being parsed as binary expressions instead
+                        //of just identifiers as is done today. We would need to add a new Parameters entry because all of this is public :(
+                        //or we could invoke the parser directly on these strings as is done for custom node symbols...
```

Comment:
Can you explain this comment?

Reply:
```
        private static FunctionDefinitionNode GenerateBuiltInMethodSignatureNode(Lang.BuiltInMethods.BuiltInMethod method)
        {
            FunctionDefinitionNode fDef = new FunctionDefinitionNode();
            fDef.Name = Lang.BuiltInMethods.GetMethodName(method.ID);
            fDef.ReturnType = method.ReturnType;
            fDef.IsExternLib = true;
            fDef.IsBuiltIn = true;
            fDef.BuiltInMethodId = method.ID;
            fDef.Signature = new ArgumentSignatureNode();
            fDef.MethodAttributes = method.MethodAttributes;

            foreach (KeyValuePair<string, Type> param in method.Parameters)
            {
                VarDeclNode arg = new VarDeclNode();
                arg.NameNode = new IdentifierNode { Name = param.Key, Value = param.Key };
                arg.ArgumentType = param.Value;
                fDef.Signature.AddArgument(arg);
            }

            return fDef;
        }
        ```

These parameter strings are always just treated like identifiers, they don't get parsed into a full AST subtree where assignment to default values would normally be supported for function signatures.